contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -591,6 +591,23 @@ saved. For more details on how to use Merge Patch, see the RFC.
 detailed explanation of how it works and why it needed to be introduced, see
 [here](/contributors/devel/sig-api-machinery/strategic-merge-patch.md).
 
+#### Declaring Patch Strategy and Object Topology
+In the context of [strategic merge patch](/contributors/devel/sig-api-machinery/strategic-merge-patch.md), a set of go markers can be used by API authors to declare the patch strategy of lists, maps and structs (`//+patchStrategy` and `//+patchMergeKey`). Work to move the `apply` functionality to the control plane ([KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/0006-apply.md)) has also introduced markers to describe the topology of lists, maps and structs (`//+listType`, `//+listMapKeys`, `//+mapType`, `//+structType`).
+
+| Marker | Definition |
+|--------|------------|
+| `//+patchStrategy=merge` |  Defines that the merge strategy will be to merge elements of the two lists into one. |
+| `//+patchStrategy=replace` (default)|  Defines that the merge strategy will be touse the list provided in the patch literally rather than merging. |
+| `// +patchMergeKey=<key-name>` |  Applicable in the context where lists are used as maps/dictionaries, i.e. where each list entry is a key-value pair. `patchMergeKey` defines which field should be seen as the key. Combined with `patchStrategy`, this describes what changes should take place on specific elements of a list as part of a patch operation. |
+| `//+listType=atomic` |  Applicable to objects that are of type list. A single actor owns and can replace the entire list, but cannot change individual elements of it. |
+| `//+listType=set` |  Applicable to objects that are of type list, and the elements are unique and scalar. Different actors can update individual elements in the list. |
```

Comment:
I think the restriction for set is that elements have to be unique and atomic (scalar or atomic list/struct/map).

Reply:
I think I agree with that restriction. But, I think currently we only correctly handle the case of scalar set elements, and have a TODO in the structured-merge-diff repo to support any atomic type as an element.

Reply:
https://github.com/kubernetes-sigs/structured-merge-diff/blob/9818ffe7e8e1c6765e6484b9ce087c8656cc60cc/typed/helpers.go#L210